libs/utils/ldif_parser.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+/**
+    * Interogate the attribute array by his type to see if exist
+    *
+    * @function alreadyExist
+    * @param {array} attributeObjectArray
+    * @param {string} type
+    * @return {object} The function will return an object if the type exist.
+    * Undefined otherwise
+    */
+function alreadyExist(attributeObjectArray, type) {
+  const lengthArray = attributeObjectArray.length;
+  for (let j = 0; j < lengthArray; j += 1) {
+    if (attributeObjectArray[j].type === type) {
+      return attributeObjectArray[j];
+    }
+  }
+  return undefined;
+}
+
+/**
+    * Return the object of an Attribute.
+    *
+    * @function objectLDAPAttribute
+    * @param {string} type
+    * @param {string} value
+    * @return {object} The function will return an object of attribute
+    */
+
+function objectLDAPAttribute(type, value) {
+  const attributeObject = ({
+    type: '',
+    value: [],
+  });
+
+  if (value === undefined || type === undefined) {
+    throw new Error('The string is not a LDAP LDIF structure');
+  }
+
+  attributeObject.type = type.trim();
+  attributeObject.value.push(value.trim());
+
+  return attributeObject;
+}
+
+/**
+    * Take an entry of LDAP and put it into an object.
+    *
+    * @function objectLDAPEntry
+    * @param {string} LDAPentry
+    * @return {object} The function will return an object of given entry
+    */
+
+function objectLDAPEntry(LDAPentry) {
+  const entryElementArray = LDAPentry.split('\n');
+  const entryElementArrayLenght = entryElementArray.length;
+  let i;
+
+  const entryObject = ({
+    dn: '',
+    attribute: [],
+  });
+
+  for (i = 0; i < entryElementArrayLenght; i += 1) {
+    const attributeArray = entryElementArray[i].split(':');
+    const type = attributeArray[0];
+    const value = attributeArray[1];
+
+    // Verify if is an attribute or the DN
+    if (type !== '' && type !== undefined) {
+
+      const currentType = alreadyExist(entryObject.attribute, type);
+
+      // Verify if the type alreadyExist
+      if (currentType) {
+        currentType.value.push(value);
+      } else {
+        const attributeResult = objectLDAPAttribute(type, value);
+        entryObject.attribute.push(attributeResult);
+      }
+
+    } else if (type !== undefined && value !== undefined) {
+
+      entryObject.dn = value.trim();
+    }
+
+  }
+  return entryObject;
+
+}
+
+/**
+  * @module LDAPtranzition
+  * @class stringJSON
+  */
+
+class stringJSON {
```

Comment:
I have to admit that I don't like this kind of parsing.
Using manual parsing forces us to think of all the possible ways people can write LDIFs and I already found a woring LDIF that breaks this parser.

The approach from the ldif library is better imo.
It uses a grammar to work with the issue. The basic flaw is that it allows to use invalid characters which breaks international necks.

I gave it a go and tried to add Unicode grammar to the peg parser. So instead of writing something cryptic as "\xFD at end of line" it write things like "ß at end of line" which seems better but still fails.

Patching the ldif library sounds like the better approach to me.

Here's the LDIF I used to break the parsing:
```
dn: cn=manfred,ou=people,o=panzerknackerag
street: Steegerstrasse 17
crypto: {"key":"Its magic, joel"}
```

Reply:
Hey Phillip I have a question there are LDIF's with the forme:
crypto: {"key":"Its magic, joel"} that as as string ->"?

Reply:
You mean an attribute with the value `{"key":"Its magic, joel"}`?
Yes. We have it a few times in LDAP.

Reply:
Ok I modify the node-ldif to accept this kind of stuff should work now I send you an email regarding this you can check that out.